docker/_include.sh

```diff
@@ -1,25 +1,26 @@
 DOCKER_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-if [ `uname` == 'Linux' -o `uname` == 'Darwin' ]; then
+if [ `uname` == 'Linux' ]; then
     XDG_DATA_HOME=${XDG_DATA_HOME:-$HOME/.local/share}
     DOCKER_DATA_HOME=$XDG_DATA_HOME/dockerhq
+    UDO="sudo"
 else
     DOCKER_DATA_HOME=/data
+    UDO=""
 fi
 
-mkdir -p $DOCKER_DATA_HOME
+if [ `uname` == 'Darwin' ]; then
+    docker-machine ssh $DOCKER_MACHINE_NAME sudo mkdir -p $DOCKER_DATA_HOME
```

Comment:
On Mac `DOCKER_DATA_HOME` is a directory inside the boot2docker VM, so be warned that if you delete your VM you'll lose all the data.

For performance reasons we probably do not want to put it in a shared folder that would locate it outside the VM on the host Mac filesystem. From what I've read that can be super slow. Maybe native Docker on Mac will make it possible?